addons/mail/static/tests/discuss_app/sidebar.test.js

```diff
@@ -64,6 +64,24 @@ test("toggling category button does not hide active category items", async () =>
     await contains(".o-mail-DiscussSidebarChannel.o-active");
 });
 
+test("toggling category button does not hide active sub thread", async () => {
+    const pyEnv = await startServer();
+    const mainChannelId = pyEnv["discuss.channel"].create({ name: "Main Channel" });
+    const subChannelId = pyEnv["discuss.channel"].create({
+        name: "Sub Channel",
+        parent_channel_id: mainChannelId,
+    });
+    await start();
+    await openDiscuss(subChannelId);
+    await contains(".o-mail-DiscussSidebarChannel");
+    await contains(".o-mail-DiscussSidebarSubchannel");
+    await click(
+        ":nth-child(1 of .o-mail-DiscussSidebarCategory) .o-mail-DiscussSidebarCategory-icon"
+    );
+    await contains(".o-mail-DiscussSidebarChannel");
+    await contains(".o-mail-DiscussSidebarSubchannel");
```

Comment:
I don't like `contains()` on just classnames, they make the test hard to read and heavily coupled with implementation details.

End-users see actual text or icons. Classnames are sometimes fine for core components like `.o-mail-Message`, but try as much as possible to not use them: they refer to technical details and test coverage on them only means the code pass if we put it this classname in DOM, which doesn't tell much.

Suggestion:
```js
await contains(".o-mail-DiscussSidebar-item", { text: "Main Channel" });
await contains(".o-mail-DiscussSidebar-item", { text: "Sub Channel" });
await click(".o-mail-DiscussSidebar button", { text: "Channels" });
await contains(".o-mail-DiscussSidebar-item", { text: "Main Channel" });
await contains(".o-mail-DiscussSidebar-item", { text: "Sub Channel" });
```